### DISCUSSION TITLE
[In progress] provide ioBroker "Bridged Devices" Adapter

### DISCUSSION THREAD

**Discussion by Apollon77** [mfucci/node-matter#57] [Workstream]
ioBroker is a Smart home aggregator written in Node.js. I would like to provide ioBroker devices as Bridged Matter Devices in a first run.

Contributors:
* @Apollon77

**Comment by vves** [comment]
@Apollon77 I am working on something similar, but to bridge some non-Matter specialty devices.  Is there a branch you are working on that I an sync with?

**Reply by Apollon77** [reply]
Honestly, please ewait 1-2 weeke until I did the new High level api ... else you need to redo anything you do now

**Reply by Hedda** [reply]
> I am working on something similar, but to bridge some non-Matter specialty devices.

FYI, the [Connectivity Standards Alliance](https://csa-iot.org/) (CSA) and CSA-IOT members have many times and are still trying to highlight that Matter bridging using multi-protocol bridges/hubs/gateways will, at least now in the beginning, be extremely important in the next upcoming years in order to allow users to connect existing non-Matter devices such as example Zigbee, Z-Wave, and Bluetooth Low Energy devices, or proprietary protocols, to your Matter fabric. Check out these very related articles:

https://csa-iot.org/newsroom/why-bridging-matters/

https://blog.espressif.com/matter-bridge-for-non-matter-devices-d3b7f003a004

https://z-wavealliance.org/matter-1-0-is-here-and-z-wave-is-accelerating-which-is-why-bridging-will-matter-to-both-protocols-and-smart-home-consumers-everywhere/

https://www.silabs.com/blog/bridging-non-matter-devices-to-a-matter-network

https://www.dusuniot.com/blog/bridge-zigbee-z-wave-devices-to-matter-gateway/

![image](https://user-images.githubusercontent.com/6320001/229043717-e110508f-016a-444b-9735-2991b8d8e899.png)

![image](https://user-images.githubusercontent.com/6320001/229053635-33f2fa41-3572-4df5-be50-4a2da6c90eaa.png)

![image](https://user-images.githubusercontent.com/6320001/229053886-ef549237-5fc4-478e-bbf9-932e54f4bc7d.png)

Has anyone given more thought to how node-matter can and will be used in a standard way to create 'whatever' to Matter bridges?

BTW, I have also seen bridges referred to as a "Matter Dynamic Bridge" or "Matter Dynamic Controller" when using dynamic endpoints:

https://github.com/project-chip/connectedhomeip/tree/master/examples/dynamic-bridge-app

https://github.com/project-chip/connectedhomeip/blob/master/examples/dynamic-bridge-app/linux/README.md

Look that it was discussed here:

https://devzone.nordicsemi.com/f/nordic-q-a/94728/matter-dynamic-controller---multiple-client-endpoints-on-one-device
